### ISSUE TITLE
Fix errors in bam_to_h5.R

### ISSUE THREAD

**Issue by ewallace** [riboviz/RiboViz#479]
@swinterbourne noticed errors arising in bam_to_h5.R:

-`ReadsToCountMatrix` had some errors putting data in the matrix. We tracked this to a bug/error in calculation of `gene_length` from `bam_file` instead of from `gene_location`. Pull request needed.

- `BamToH5` throws an error writing. It's probably a change in the functions to write data in rhdf5.
```
Error: 'h5writeAttribute.integer' is not an exported object from 'namespace:rhdf5'
```
To do: read the manual for `h5write` and see what we should change. https://rdrr.io/bioc/rhdf5/man/h5write.html

Hypothesis: replacing `h5writeAttribute.integer` (not exported) by `h5writeAttribute` (exported) will fix the problem.

- After fixing the error, I (@ewallace) should run the integration tests to check the vignette run pretty much the same way.

**Comment by acope3** [comment]
@ewallace @swinterbourne see #477

**Comment by ewallace** [comment]
This was fixed by #480
